commit dce1ade8a5bf2beb99646ceff44991ed59c0e4d3
Author: Dosakayala Nagalakshmi <nadosakayala@synamedia.com>
Date:   Mon Aug 25 10:27:30 2025 +0530

    Implementation changes hdmiin

diff --git a/build_dependencies.sh b/build_dependencies.sh
index 090f105..566fca2 100644
--- a/build_dependencies.sh
+++ b/build_dependencies.sh
@@ -47,7 +47,7 @@ cd rfc
 autoreconf -i
 ./configure --enable-rfctool=yes --enable-tr181set=yes
 cd rfcapi
-make librfcapi_la_CPPFLAGS="-I/usr/include/cjson"
+make CXXFLAGS="-DUSE_IARMBUS" librfcapi_la_CPPFLAGS="-I/usr/include/cjson"
 make install
 export RFC_PATH=$ROOT/rfc
 
diff --git a/ds/host.cpp b/ds/host.cpp
index d621e47..af63b13 100644
--- a/ds/host.cpp
+++ b/ds/host.cpp
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2016 RDK Management
+ * Copyright 2025 RDK Management
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -241,7 +241,6 @@ namespace device
         }
     }
 
-
 /**
  * @fn bool Host::setPowerMode(int mode)
  * @brief This API is used to change the power mode of the device.
@@ -913,6 +912,145 @@ namespace device
     printf ("%s:%d - Set Audio Mixer levels for audio input: %d with volume = %d\n", __PRETTY_FUNCTION__, __LINE__,aInput, volume);
    }
 
+/**
+ * @fn void  Host::Register(IHDMIInEvent *listener)
+ * @brief This API is used to register the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(IHDMIInEvent *listener) 
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::UnRegister(IHDMIInEvent *listener)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(IHDMIInEvent *listener) 
+{
+  return 0;
+}
+
+/**
+ * @fn void  Host::Register(ICompositeInEvent *Evtnotification)
+ * @brief This API is used to register the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(ICompositeInEvent *listener)
+{
+  return 0;
+}
+
+/**
+ * @fn void  Host::UnRegister(ICompositeInEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(ICompositeInEvent *listener)
+{
+  return 0;
+}
+
+/**
+ * @fn void  Host::UnRegister(IDisplayHDMIHotPlugEvent *Evtnotification)
+ * @brief This API is used to Register the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(IDisplayHDMIHotPlugEvent *listener)
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::UnRegister(IDisplayHDMIHotPlugEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(IDisplayHDMIHotPlugEvent *listener)
+{
+  return 0;
+}
+
+/**
+ * @fn void  Host::UnRegister(IDisplayEvent *Evtnotification)
+ * @brief This API is used to Register the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(IDisplayEvent *listener)
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::UnRegister(IDisplayEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(IDisplayEvent *listener)
+{
+  return 0;
+}
+
+/**
+ * @fn void  Host::Register(IAudioOutputPortEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(IAudioOutputPortEvent *listener)
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::UnRegister(IAudioOutputPortEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(IAudioOutputPortEvent *listener)
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::Register(IVideoDeviceEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::Register(IVideoDeviceEvent *listener)
+{
+  return 0;
+}
+
+
+/**
+ * @fn void  Host::UnRegister(IVideoDeviceEvent *Evtnotification)
+ * @brief This API is used to UnRegister the Events
+ *
+ * @return unint32_t
+ */
+uint32_t Host::UnRegister(IVideoDeviceEvent *listener)
+{
+  return 0;
+}
+
+
 
 }
 
diff --git a/ds/include/host.hpp b/ds/include/host.hpp
index d14d42d..f7997f5 100644
--- a/ds/include/host.hpp
+++ b/ds/include/host.hpp
@@ -2,7 +2,7 @@
  * If not stated otherwise in this file or this component's LICENSE file the
  * following copyright and licenses apply:
  *
- * Copyright 2016 RDK Management
+ * Copyright 2025 RDK Management
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -17,8 +17,6 @@
  * limitations under the License.
 */
 
-
-
 /**
 * @defgroup devicesettings
 * @{
@@ -38,6 +36,9 @@
 #include "videoDevice.hpp"
 #include "sleepMode.hpp"
 #include  "list.hpp"
+#include "dsMgr.h"
+#include "dsTypes.h"
+#include "dsDisplay.h"
 
 #include  <list>
 #include <string>
@@ -59,6 +60,222 @@ namespace device {
  */
 class Host {
 public:
+       
+    struct IHDMIInEvent {
+            
+	    // @brief HDMI Event Hot Plug
+            // @text onHDMIInEventHotPlug
+            // @param port: port 0 or 1 et al
+            // @param isConnected: is it connected (true) or not (false)
+            virtual void OnHDMIInEventHotPlug(dsHdmiInPort_t port, bool isConnected) { };
+ 
+            // @brief HDMI Event Signal status
+            // @text OnHDMIInEventSignalStatus
+            // @param port: port 0 or 1 et al
+            // @param signalStatus: Signal Status
+            virtual void OnHDMIInEventSignalStatus(dsHdmiInPort_t port, dsHdmiInSignalStatus_t signalStatus) { };
+ 
+            // @brief HDMI Event Signal status
+            // @text onHDMIInEventStatus
+            // @param activePort: port 0 or 1 et al
+            // @param isPresented: is it presented or not
+            virtual void OnHDMIInEventStatus(dsHdmiInPort_t activePort, bool isPresented) { };
+ 
+            // @brief HDMI Video Mode update
+            // @text onHDMInVideoModeUpdate
+            // @param port: port 0 or 1 et al
+            // @param videoPortResolution: Video port resolution
+            virtual void OnHDMInVideoModeUpdate(dsHdmiInPort_t port, const dsVideoPortResolution_t& videoPortResolution) { };
+ 
+            // @brief HDMI ALLM (Auto Low Latency Mode) status
+            // @text onHDMInAllmStatus
+            // @param port: port 0 or 1 et al
+            // @param allmStatus: allm status
+            virtual void OnHDMInAllmStatus(dsHdmiInPort_t port, bool allmStatus) { };
+ 
+            // @brief HDMI Event AVI content type
+            // @text OnHDMInAVIContentType
+            // @param port: port 0 or 1 et al
+            // @param aviContentType: AVI content type
+            virtual void OnHDMInAVIContentType(dsHdmiInPort_t port, dsAviContentType_t aviContentType) { };
+ 
+            // @brief HDMI VRR status
+            // @text OnHDMInVRRStatus
+            // @param port: port 0 or 1 et al
+            // @param vrrType: VRR type
+            virtual void OnHDMInVRRStatus(dsHdmiInPort_t port, dsVRRType_t vrrType) { };
+
+	    // @brief Zoom settings changed
+            // @text OnZoomSettingsChanged
+            // @param zoomSetting: Currently applied zoom setting
+            virtual void OnZoomSettingsChanged(dsVideoZoom_t zoomSetting) { };
+    };
+
+    uint32_t Register(IHDMIInEvent *listener);
+    uint32_t UnRegister(IHDMIInEvent *listener);
+
+
+    struct ICompositeInEvent {
+            // @brief Composite In Hotplug event
+            // @text onCompositeInHotPlug
+            // @param port: Port of the hotplug
+            // @param isConnected: Is it connected (true) or not(false)
+            virtual void OnCompositeInHotPlug(dsCompositeInPort_t port, bool isConnected) { };
+ 
+            // @brief Composite In Signal status
+            // @text onCompositeInSignalStatus
+            // @param port: Port of the hotplug
+            // @param signalStatus: Signal status
+            virtual void OnCompositeInSignalStatus(dsCompositeInPort_t port, dsCompInSignalStatus_t signalStatus) { };
+ 
+            // @brief Composite In status
+            // @text onCompositeInStatus
+            // @param activePort: Active port
+            // @param isPresented: is it presented to user
+            virtual void OnCompositeInStatus(dsCompositeInPort_t activePort, bool isPresented) { };
+
+
+            // @brief Composite In Video Mode Update
+            // @text OnCompositeInVideoModeUpdate
+            // @param activePort: Active port
+            // @param videoResolution: See DisplayVideoPortResolution
+            virtual void OnCompositeInVideoModeUpdate(dsCompositeInPort_t activePort, dsVideoPortResolution_t videoResolution) { };
+    };
+
+    uint32_t Register(ICompositeInEvent *listener);
+    uint32_t UnRegister(ICompositeInEvent *listener);
+    
+
+    struct IDisplayHDMIHotPlugEvent{
+
+            // @brief Display HDMI Hot plug event
+            // @text onDisplayHDMIHotPlug
+            // @param displayEvent: DS_DISPLAY_EVENT_CONNECTED or DS_DISPLAY_EVENT_DISCONNECTED
+            virtual void OnDisplayHDMIHotPlug(dsDisplayEvent_t displayEvent) { };
+    };
+    
+    uint32_t Register(IDisplayHDMIHotPlugEvent*listener);
+    uint32_t UnRegister(IDisplayHDMIHotPlugEvent *listener);
+
+
+
+    struct IDisplayEvent{
+
+            // @brief Display RX Sense event
+            // @text onDisplayRxSense
+            // @param displayEvent: DS_DISPLAY_RXSENSE_ON or DS_DISPLAY_RXSENSE_OFF
+            virtual void OnDisplayRxSense(dsDisplayEvent_t displayEvent) { };
+ 
+            // @brief Display HDCP Status
+            // @text OnDisplayHDCPStatus
+            virtual void OnDisplayHDCPStatus() { };
+    };
+    
+    uint32_t Register(IDisplayEvent *listener);
+    uint32_t UnRegister(IDisplayEvent *listener);
+
+
+    struct IAudioOutputPortEvent{
+
+            // @brief Associated Audio mixing changed
+            // @text onAssociatedAudioMixingChanged
+            // @param mixing: true or false
+            virtual void OnAssociatedAudioMixingChanged(bool mixing) { };
+    
+            // @brief Audio Fader balance changed
+            // @text onAudioFaderControlChanged
+            // @param mixerBalance: applied mixer balance value
+            virtual void OnAudioFaderControlChanged(int mixerBalance) { };
+    
+            // @brief Primary language for Audio changed
+            // @text onAudioPrimaryLanguageChanged
+            // @param primaryLanguage: current primary language for audio
+            virtual void OnAudioPrimaryLanguageChanged(const std::string& primaryLanguage) { };
+
+            // @brief Secondary language for Audio changed
+            // @text onAudioSecondaryLanguageChanged
+            // @param secondaryLanguage: current secondary language for audio
+            virtual void OnAudioSecondaryLanguageChanged(const std::string& secondaryLanguage) { };
+    
+            // @brief Audio output hot plug event
+            // @text onAudioOutHotPlug
+            // @param portType: Type of audio port see AudioPortType
+            // @param uiPortNumber: The port number assigned by UI
+            // @param isPortConnected: true (connected) or false (not connected)
+            virtual void OnAudioOutHotPlug(dsAudioPortType_t  audioPortType, int uiPortNumber, bool isPortConnected) { };
+
+
+            // @brief Dolby Atmos capabilities changed
+            // @text onDolbyAtmosCapabilitiesChanged
+            // @param atmosCapability: the dolby atmos capability
+            // @param status: true (available) or false (not available)
+            virtual void OnDolbyAtmosCapabilitiesChanged(dsATMOSCapability_t atmosCapability, bool status) { };
+
+            // @brief Audio port state changed
+            // @text onAudioPortStateChanged
+            // @param audioPortState: audio port state
+            virtual void OnAudioPortStateChanged(dsAudioPortState_t audioPortState) { };
+
+            // @brief Audio mode for the respective audio port - raised for every type of port
+            // @text onAudioModeEvent
+            // @param audioPortType: audio port type see AudioPortType
+            // @param audioMode: audio mode - see audioStereoMode
+            virtual void OnAudioModeEvent(dsAudioPortType_t  audioPortType, dsAudioStereoMode_t audioMode) { };
+
+            // @brief Audio Output format changed
+            // @text onAudioFormatUpdate
+            // @param audioFormat: Type of audio format see AudioFormat
+            virtual void OnAudioFormatUpdate(dsAudioFormat_t audioFormat) { };
+
+    };
+
+    uint32_t Register(IAudioOutputPortEvent *listener);
+    uint32_t UnRegister(IAudioOutputPortEvent *listener);
+
+
+    struct IVideoDeviceEvent {
+            // @brief Display Framerate Pre-change
+            // @text OnDisplayFrameratePreChange
+            // @param frameRate: PreChange framerate
+            virtual void OnDisplayFrameratePreChange(const std::string& frameRate) { };
+
+            // @brief Display Framerate Post-change
+            // @text OnDisplayFrameratePostChange
+            // @param frameRate:  framerate post change
+            virtual void OnDisplayFrameratePostChange(const std::string& frameRate) { };
+     };
+        
+     uint32_t Register(IVideoDeviceEvent *listener);
+     uint32_t UnRegister(IVideoDeviceEvent *listener);
+	
+
+     struct IVideoOutputPortEvent {
+            
+	    // @brief On Resolution Pre changed
+            // @text OnResolutionPreChange
+            // @param resolution: resolution
+            virtual void OnResolutionPreChange(const int width, const int height) { };
+ 
+            // @brief On Resolution Post change
+            // @text onResolutionPostChange
+            // @param resolution: resolution
+            virtual void OnResolutionPostChange(const int width, const int height) { };
+ 
+            // @brief On HDCP Status change
+            // @text OnHDCPStatusChange
+            // @param hdcpStatus: HDCP Status
+            virtual void OnHDCPStatusChange(dsHdcpStatus_t hdcpStatus) { };
+ 
+            // @brief On Video Format update
+            // @text OnVideoFormatUpdate
+            // @param videoFormatHDR: Video format HDR standard
+            virtual void OnVideoFormatUpdate(dsHDRStandard_t videoFormatHDR) { };
+    };
+
+    uint32_t Register(IVideoOutputPortEvent *listener);
+    uint32_t UnRegister(IVideoOutputPortEvent *listener);
+
+
     static const int kPowerOn;
     static const int kPowerOff;
     static const int kPowerStandby;
